code/modules/clothing/glasses/thermal.dm

```diff
@@ -0,0 +1,90 @@
+/obj/item/clothing/glasses/thermal
+	name = "Optical Thermal Scanner"
+	desc = "Thermals in the shape of glasses."
+	icon_state = "thermal"
+	item_state = "glasses"
+	action_button_name = "Toggle Optical Matrix"
+	origin_tech = list(TECH_MAGNET = 3)
+	toggleable = 1
+	prescription = 1
+	vision_flags = SEE_MOBS
+	see_invisible = SEE_INVISIBLE_NOLIGHTING
+	flash_protection = FLASH_PROTECTION_REDUCED
+	active = 0
+	var/tick_cost = 0.5
+	var/obj/item/weapon/cell/cell = null
+	var/suitable_cell = /obj/item/weapon/cell/small
+
+	emp_act(severity)
+		if(ishuman(src.loc))
+			var/mob/living/carbon/human/M = src.loc
+			M << "<span class='danger'>The Optical Thermal Scanner overloads and blinds you!</span>"
```

Comment:
Looks like this output should be after next 'if' condition. Also please use [src.name] instead of object name hardcoding.